src/ddmd/mars.d

```diff
@@ -849,6 +850,31 @@ private int tryMain(size_t argc, const(char)** argv)
             printf("%.*s", cast(int)ob.offset, ob.data);
     }
 
+    if(global.params.compileimports && global.params.link)
+    {
+        auto importedModulesToCompile = Modules();
+        fprintf(global.stdmsg, "compileimports: start\n");
+        for (size_t i = 0; i < Module.amodules.dim; i++)
```

Comment:
Don't be afraid of using D features - the current codebase just doesn't in many places because it was automatically transpiled from C++

`foreach(i; 0 .. Module.amodules.dim)`

Reply:
I did a search in all of ddmd and found like 1 or 2 instances of `foreach` which really surprised me.  Good to know it's ok to use the D feature in this case.